### PR TITLE
feat: 🎸 Allow passing of `nonce` as part of `ProcedureOptions`

### PR DESCRIPTION
**Pull request by prashantasdeveloper**
### Description

Add `nonce` parameter to `ProcedureOptions`

### JIRA Link

DA-154

### Checklist

- [ ] Updated the Readme.md (if required) ?

### CODE REVIEW

**Review by polymath-eric — APPROVED** [review]
LGTM

**Review by monitz87 — COMMENTED** [review]
I just realized that this won't work properly until we eliminate Transaction Queues. Every transaction in the queue would have the same nonce with this implementation. If we were to take the passed value and increment it for every tx in the queue, users trying to run multiple procedures at once wouldn't know how much to increment it by unless they prepare each one in sequence and count the amount of transactions, which seems like very bad UX

Once we eliminate TQs it becomes much easier

File: src/types/index.ts [hunk 1]
```diff
@@ -1284,6 +1284,13 @@ export interface ProcedureOpts {
    * Account or address of a signing key to replace the current one (for this procedure only)
    */
   signingAccount?: string | Account;
+
+  /**
+   * nonce value for signing the transaction
+   *
+   * More information can be found at: https://polkadot.js.org/docs/api/cookbook/tx/#how-do-i-take-the-pending-tx-pool-into-account-in-my-nonce
+   */
+  nonce?: BigNumber;
```
Comment (polymath-eric) [inline src/types/index.ts:1293]: We need to expose a method to get the nonce of an `Account`. I think a method like `currentNonce` there would be reasonable. I don't think there is a way to get that now, which makes this hard to use.

Then for this field we can link too that method instead about getting the current nonce for the signingAccount. 


Also, what do you think about having this be `BigNumber | () => BigNumber`  or even `() => BigNumber`. By giving a function a user could call `currentNonce` and make a closure and be able to increment it. All we'd have to do is call it and use the result.

For example:
```ts
  let nonce = 0
  const incNonce = (n) => () => (nonce += n);
  const addOne = incNonce(1);

  console.log(addOne());  // 1
  console.log(addOne()); // 2
  console.log(addOne()); // 3
  console.log(nonce); // 3
```

Comment (monitz87) [reply src/types/index.ts:1293]: Agree on everything. I would leave the option of passing a plain number as well as a function

File: src/types/index.ts [hunk 2]
```diff
@@ -1288,9 +1288,9 @@ export interface ProcedureOpts {
   /**
    * nonce value for signing the transaction
    *
-   * More information can be found at: https://polkadot.js.org/docs/api/cookbook/tx/#how-do-i-take-the-pending-tx-pool-into-account-in-my-nonce
+   * An {@link api/entities/Account!Account} can directly fetch its current nonce by calling {@link api/entities/Account!Account.getCurrentNonce | account.getCurrentNonce}. More information can be found at: https://polkadot.js.org/docs/api/cookbook/tx/#how-do-i-take-the-pending-tx-pool-into-account-in-my-nonce
```
Comment (monitz87) [inline src/types/index.ts:1291]: I would add a section explaining how to use it with a function

**Review by monitz87 — CHANGES_REQUESTED** [review]

**Review by polymath-eric — APPROVED** [review]
LGTM. 

Sonar Cloud lists two smells BTW

File: src/types/index.ts [hunk 3]
```diff
@@ -1289,8 +1289,10 @@ export interface ProcedureOpts {
    * nonce value for signing the transaction
    *
    * An {@link api/entities/Account!Account} can directly fetch its current nonce by calling {@link api/entities/Account!Account.getCurrentNonce | account.getCurrentNonce}. More information can be found at: https://polkadot.js.org/docs/api/cookbook/tx/#how-do-i-take-the-pending-tx-pool-into-account-in-my-nonce
+   *
+   * @note the passed value can be either the nonce itself or a function that returns the nonce. This allows, for example, passing a closure that increases the returned value every time it's called, or a function that fetches the nonce from the chain or something else
```
Comment (monitz87) [inline src/types/index.ts:1293]:
```suggestion
   * @note the passed value can be either the nonce itself or a function that returns the nonce. This allows, for example, passing a closure that increases the returned value every time it's called, or a function that fetches the nonce from the chain or a different source
```

**Review by monitz87 — APPROVED** [review]

**Pull request merged**